src/constructor.js

```diff
@@ -4,25 +4,42 @@ function Todo(index, completed, description) {
   this.description = description;
 }
 
-function TodoList() {
-  this.list = [];
-  this.new = (completed, description) => {
+class TodoList {
+  constructor() {
+    this.list = JSON.parse(localStorage.getItem('TodoList')) || [new Todo(1, false, 'Click on the text to edit'),
+      new Todo(2, false, 'Drag \'n drop to reorder your list'),
+      new Todo(3, false, 'Manage all your lists in one place'),
+      new Todo(4, false, 'Resync to clear out the old')];
+  }
+
+  new(completed, description) {
     this.list.push(new Todo(this.list.length + 1, completed, description));
-  };
-  this.delete = (indexInput) => {
+    this.save();
+  }
+
+  delete(indexInput) {
     const index = this.list.findIndex((item) => item.index === parseInt(indexInput, 10));
     this.list.splice(index, 1);
-  };
-  this.swap = (indexa, indexb) => {
+    this.save();
+  }
+
+  swap(indexa, indexb) {
     const x = this.list.findIndex((item) => item.index === parseInt(indexa, 10));
     const y = this.list.findIndex((item) => item.index === parseInt(indexb, 10));
-    const des = this.list[x].description;
-    const comp = this.list[x].completed;
-    this.list[x].completed = this.list[y].completed;
-    this.list[x].description = this.list[y].description;
-    this.list[y].completed = comp;
-    this.list[y].description = des;
-  };
+    const todo1 = this.list[x];
+    this.list[x] = this.list[y];
+    this.list[y] = todo1;
+    this.save();
+  }
+
+  save() {
+    localStorage.setItem('TodoList', JSON.stringify(this.list));
+  }
+
+  completed(index, value) {
+    this.list.find((item) => item.index === parseInt(index, 10)).completed = value;
+    this.save();
+  }
```

Comment:
- [ ] when I click the checkbox, the value doesn't change to true, I saw that your default is true, your default should be false on the creation of a new task and toggles between true and false.

Kindly check your implementation again

<img width="1667" alt="Screenshot 2021-08-18 at 20 14 26" src="https://user-images.githubusercontent.com/34798898/129958305-4337070b-29ab-46c9-bc6c-a6d1095ceba0.png">